x-pack/plugins/apm/server/routes/data_view/create_static_data_view.ts

```diff
@@ -57,16 +57,40 @@ export async function createStaticDataView({
         savedObjectsClient.create(
           'index-pattern',
           {
-            ...apmDataView.attributes,
+            // required fields (even if empty)
             title: apmDataViewTitle,
+            fieldAttrs: '{}',
+            fields: '[]',
+            runtimeFieldMap: '{}',
+            timeFieldName: '@timestamp',
+            typeMeta: '{}',
+
+            // link to APM from Discover
+            fieldFormatMap: JSON.stringify({
+              [TRACE_ID]: {
+                id: 'url',
+                params: {
+                  urlTemplate: 'apm/link-to/trace/{{value}}',
+                  labelTemplate: '{{value}}',
+                },
+              },
+              [TRANSACTION_ID]: {
+                id: 'url',
+                params: {
+                  urlTemplate: 'apm/link-to/transaction/{{value}}',
+                  labelTemplate: '{{value}}',
+                },
+              },
+            }),
```

Comment:
Replace the fields from `index_pattern.json` with the required fields, and custom field formatting.